File: src/modules/organization/validators/organization_existing_validator.py
from src.modules.organization.models import Organization
from src.shared.exceptions.not_found_exception import NotFoundException


def organization_existing_validator(organization_id: int):

    organization = Organization.objects.filter(
        id=organization_id, deleted_at=None
    ).first()

    if organization is None:
        raise NotFoundException("Organização não encontrada!")

File: src/modules/organization/validators/organization-existing.validator.ts
import { NotFoundException } from '@nestjs/common';
import { PrismaService } from 'src/infra/database/prisma.service';

export async function organizationExistingValidator(organizationId: number) {
  const prisma = new PrismaService();

  const organization = await prisma.organization.findFirst({
    where: {
      id: organizationId,
      deletedAt: null,
    },
  });

  if (!organization) throw new NotFoundException('Organização não encontrada!');
}
